docs/content/Schema/Fundamentals/Concepts.md

```diff
@@ -0,0 +1,303 @@
+cube('Orders', {
+  sql: `
+SELECT
+  *
+FROM
+  orders,
+  line_items
+WHERE
+  orders.id = line_items.order_id
+  `,
+});
+```
+
+## Dimensions
+
+Dimensions represent the properties of a **single** data point in the cube.
+[The `orders` table](#top) contains only dimensions, so representing them in the
+`Orders` cube is straightforward:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  dimensions: {
+    id: {
+      sql: `id`,
+      type: `number`,
+      // Here we explicitly let Cube.js know this field is the primary key
+      // This is required for de-duplicating results when using joins
+      primaryKey: true
+    },
+    status: {
+      sql: `status`,
+      type: `string`
+    },
+  },
+});
+```
+
+[The `line_items` table](#top) also has a couple of dimensions which can be
+represented as follows:
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  dimensions: {
+    id: {
+      sql: `id`,
+      type: `number`,
+      // Again, we explicitly let Cube.js know this field is the primary key
+      // This is required for de-duplicating results when using joins
+      primaryKey: true
+    },
+  },
+});
+```
+
+Dimensions can be of different types, and you can find them all
+[here][ref-schema-dimension-types].
+
+### Time Dimensions
+
+Time-based properties should be represented as dimensions with type `time`. Time
+dimensions allow granulating the result set for queries sent to Cube.js.
+
+We can add the necessary time dimensions to both schemas as follows:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+
+    completedAt: {
+      sql: `completed_at`,
+      type: `time`,
+    },
+  },
+});
+```
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+  },
+});
+```
+
+Time dimensions are essential to enabling performance boosts such as
+[partitioned pre-aggregations][ref-caching-use-preaggs-partition-time] and
+[incremental refreshes][ref-tutorial-incremental-preagg].
+
+## Measures
+
+Measures represent the properties of a **set of data points** in the cube. To
+add a measure called `count` to our `Orders` cube, for example, we can do the
+following:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  measures: {
+    count: {
+      type: `count`,
+    },
+  },
+});
+```
+
+In our `LineItems` cube, we can We could also create a measure to sum up the
+total value of line items sold:
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  measures: {
+    total: {
+      sql: `price`,
+      type: `sum`,
+    },
+  },
+})
+```
+
+Measures can be of different types, and you can find them all
+[here][ref-schema-measure-types].
+
+## Joins
+
+Joins define the relationships between cubes, which then allows accessing and
+comparing properties from two or more cubes at the same time. In Cube.js, all
+joins are `LEFT JOIN`s by default.
```

Comment:
I wonder if we should keep "by default" here. There's no way to change these defaults, right?